src/Event/Kafka/KafkaRecord.php

```diff
@@ -0,0 +1,64 @@
+<?php declare(strict_types = 1);
+
+namespace Bref\Event\Kafka;
+
+
+use InvalidArgumentException;
+
+final class KafkaRecord
+{
+
+    private $record;
+
+    /**
+     * @param mixed $record
+     */
+    public function __construct($record)
+    {
+        if (! is_array($record)) {
+            throw new InvalidArgumentException;
+        }
+        $this->record = $record;
+    }
+
+    public function getTopic(): string
+    {
+        return $this->record['topic'];
+    }
+
+    public function getPartition(): int
+    {
+        return $this->record['partition'];
+    }
+
+    public function getOffset(): int
+    {
+        return $this->record['offset'];
+    }
+
+    public function getTimestamp(): int
+    {
+        return $this->record['timestamp'];
+    }
+
+    public function getValue(): mixed
+    {
+        return base64_decode($this->record['value']);
+    }
+
+    public function getHeaders(): array
+    {
+        return array_map(
+            function (array $chars): string {
+                return implode('', array_map(
+                    function (int $char): string {
+                        return chr($char);
```

Comment:
Why is this behavior needed?

Reply:
Oh interesting, thanks!

Could you add that explanation in a comment in the code? That way it's there forever :p (especially useful in 2 years when we revisit that code and are surprised by it)

Reply:
I added a small explanation and a link to the javadocs. i think that should be enough.